services/speech_to_text/v1.js

```diff
@@ -74,121 +216,105 @@ module.exports = function (RED) {
+              = 'Audio format (' + f + ') not supported, must be encoded as WAV, FLAC or OGG.';
 
-        if (format === 'ogg') { 
-          format += ';codecs=opus';
+          node.error(message_err_format, msg)
+          return;  
         }
+      }  
 
-        var params = {};
-        
-        if(config.continuous === 'true')
-        {
-          params = {
-            audio: audio,
-            content_type: 'audio/' + format,
-            model: model,
-            continuous: true
-          };
-
-        } else {
-          params = {
-            audio: audio,
-            content_type: 'audio/' + format,
-            model: model,
-            continuous: false
-          };
+      // We are now ready to process the input data 
+      // If its a buffer then need to read it all before invoking the service 
+      if (msg.payload instanceof Buffer) {
+        temp.open({suffix: '.' + fileType(msg.payload).ext}, function (err, info) {
+          if (err) {
+            this.status({fill:'red', shape:'ring', text:'unable to open audio stream'});          
+            var message ='Node has been unable to open the audio stream'; 
 
-        }
+            node.error(message, msg);
+            return;        
+          }  
 
+          stream_buffer(info.path, msg.payload, function (format) {
+            var audio = fs.createReadStream(info.path);
 
-        node.status({fill:'blue', shape:'dot', text:'requesting'});
-        speech_to_text.recognize(params, function (err, res) {
-          node.status({})
-          if (err) {
-            node.error(err, msg);
-          } else {
-            msg.transcription = '';
-            if (res.results.length && res.results[0].alternatives.length) {
-              msg.transcription = res.results[0].alternatives[0].transcript;
-            }
-          }
-
-          node.send(msg);
-          if (cb) {
-            cb();
-          }
+            performAction(audio, format, actionComplete, temp.cleanup);        
+          });
         });
-      };
-
-      var stream_buffer = function (file, contents, cb) {
-        fs.writeFile(file, contents, function (err) {
+      } else if (urlCheck(msg.payload)) {
+        temp.open({suffix: '.audio'}, function(err, info){
           if (err) {
-            throw err;
-          }
-          cb(fileType(contents).ext)
-        });
-      };
-
-      var stream_url = function (file, location, cb) {
-        var wstream = fs.createWriteStream(file);
-
-        wstream.on('finish', function () {
-          fs.readFile(file, function (err, buf) {
-            if (err) {
-              throw err;
-            }
-            cb(fileType(buf).ext)
-          })
-        });
-
-        request(location)
-        .pipe(wstream);
-      };
+            this.status({fill:'red', shape:'ring', 
+              text:'unable to open url audio stream'});          
+            var message ='Node has been unable to open the url audio stream'; 
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2636145243/issues/source?bid=3225115&fileBranchId=3319654#l291)